modules/administration-guide/partials/proc_importing-extensions-that-are-not-in-the-open-vsx-registry.adoc

```diff
@@ -3,14 +3,53 @@
 [id="importing-extensions-that-are-not-in-the-open-vsx-registry"]
 = Importing extensions that are not in the Open VSX Registry
 
-In the intro, please clarify if this procedure is about adding extensions to Che's embedded Open VSX Registry.
+This section describes how to add VS Code extensions that are not in the link:https://open-vsx.org/[Open VSX Registry] to {prod-short}'s embedded Open VSX Registry.
 
-Please clarify the use cases for adding extensions that are not in the Open VSX Registry. Is this procedure about adding extensions that are in https://marketplace.visualstudio.com/VSCode? If so, then please clearly state it as the (main?) use case for this procedure.
+It's possible that link:https://open-vsx.org/[Open VSX Registry] doesn't contain an extension that needs to be used in the workspace. If the extension was not yet published on the link:https://open-vsx.org/[Open VSX Registry]. In that case, please reach out to the extension's author and ask them to publish the extension on the Open VSX registry, the link:https://open-vsx.org/[how to Publish an Extension] docs provide instructions or try to import it into the local registry by using the procedure from this page.
 
 .Prerequisites
-
-?
+. The vsix file of the VS Code extension should be deployed somewhere to download it by the direct link.
+. The extenshion's id and version that is going to be added.  
 
 .Procedure
-
-?
+. Modify `openvsx-sync.json` file in the link:https://github.com/redhat-developer/devspaces/blob/devspaces-3-rhel-8/dependencies/che-plugin-registry/openvsx-sync.json[plugin-registry repository] to add an information about the extension:
++
+[source,json]
+----
+    {
+        "id": "extension_id",
+        "download": "url_to_download_vsix_file",
+        "version": "extension_version"
+    }
+----
+for example:
++
+[source,json]
+----
+    {
+        "id": "redhat.vscode-yaml",
+        "version": "1.10.1",
+        "download": "https://github.com/redhat-developer/vscode-yaml/releases/download/1.10.1/yaml-1.10.1-19523.vsix"
+    }
+----
+NOTE: If the extension is available in link:https://marketplace.visualstudio.com/vscode[VS Code Marketplace], the download link could be build like https://extension_publisher.gallery.vsassets.io/_apis/public/gallery/publisher/extension_publisher/extension/extension_name/extension_version/assetbyname/Microsoft.VisualStudio.Services.VSIXPackage. For GoogleCloudTools.cloudcode extension it is https://GoogleCloudTools.gallery.vsassets.io/_apis/public/gallery/publisher/GoogleCloudTools/extension/cloudcode/1.20.3/assetbyname/Microsoft.VisualStudio.Services.VSIXPackage
```

Comment:
I'm not sure if it's good to have such info in our docs. 
I've seen this article https://www.eclipse.org/community/eclipse_newsletter/2020/march/1.php
and
```
Downloading extensions from the Microsoft Marketplace for any use other than in Microsoft products is prohibited as well.
```
@nickboldt @mario WDYT?

Reply:
unless we support / use this method, then +1, we shouldn't mention it.

And if it violates M$'s license terms, then +100, we should definitely remove it.

Reply:
I think that we can mention how to retrieve the download URL of an extension published in the microsoft marketplace (there is a download link in the msft marketplace anyway). We should warn that downloading the extension is prohibited but I don't see anything that confirms that in the [Microsoft Visual Studio Marketplace Terms of Use](https://aka.ms/vsmarketplace-ToU) (maybe that has changed in the meantime?).

Reply:
I definitely recall a "thou shalt only use these extensions with our products" caveat being on the site at some point, but http://aka.ms/VSMarketplace-TOU (last updated June 2021) doesn't have that anywhere. Also nothing in an older version https://cdn.vsassets.io/v/M146_20190123.39/_content/Microsoft-Visual-Studio-Marketplace-Terms-of-Use.pdf from Jan 2019. 

Maybe it was somewhere else, like on https://github.com/VSCodium/vscodium/blob/master/DOCS.md#extensions--marketplace which says "Being a vscode based editor, VSCodium gets additional features by installing VS Code extensions. Unfortunately, as Microsoft prohibits usages of the Microsoft marketplace by any other products or redistribution of .vsix files from it, in order to use VS Code extensions in non-Microsoft products those need to be installed differently." and links to https://github.com/microsoft/vscode/issues/31168#issuecomment-606757206 and https://github.com/microsoft/vscode/issues/31168#issuecomment-1136375670, which states in summary:

```
Since Microsoft won't directly interpret their Visual Studio Marketplace Terms of Use for you, I will since they're quite clear:

Short Answer: NO, YOU CANNOT LEGALLY USE THE MICROSOFT MARKETPLACE IN ANY NON-MICROSOFT PRODUCT.
```

See also https://github.com/microsoft/vscode/wiki/Differences-between-the-repository-and-Visual-Studio-Code

Reply:
Given the [whois data](https://who.is/whois/vsassets.io) for vsassets.io shows it's owned by M$, I would assume that pulling files from there is the same as pulling files from https://marketplace.visualstudio.com/ and therefore would violate their TOU. 

-1 to link to asset files. If people want to figure it out themselves and violate the TOU, that's up to them. We should not document how to make it easier.

Reply:
Ok so let's not mention the direct URL to the Microsoft marketplace @svor

Reply:
fixed in PR